### PR TITLE
[WIP] feature: first draft of azure batch subcommand

### PR DESCRIPTION
**Pull request by claymcleod**

### CODE REVIEW

**Comment by claymcleod** [comment]
There's actually a pretty obscure bug here: both `azure-batch` and `azure-cosmos` both export a module called `azure`. In my case, when trying to test out `oliver azure batch`, I get the following error:

```
Traceback (most recent call last):
  File "/Users/cmcleod/conda/bin/oliver", line 7, in <module>
    exec(compile(f.read(), __file__, 'exec'))
  File "/Users/cmcleod/Developer/stjudecloud/oliver/scripts/oliver", line 9, in <module>
    from oliver.subcommands import (
  File "/Users/cmcleod/Developer/stjudecloud/oliver/oliver/subcommands/azure.py", line 6, in <module>
    from ..integrations.azure import batch, cosmos
  File "/Users/cmcleod/Developer/stjudecloud/oliver/oliver/integrations/azure/batch.py", line 1, in <module>
    import azure.batch._batch_service_client as batch
ModuleNotFoundError: No module named 'azure.batch'
```

After digging in, this is because the `azure-cosmos` package happens to be first on my `PYTHONPATH`, so `import azure` doesn't contain anything to do with batch.

I'll let you decide how you want to handle, sorry I couldn't do more to test this out after I made changes to the new framework for azure/aws subcommands.

**Comment by adthrasher** [comment]
@claymcleod - Azure provides both an azure package and individual service packages (e.g. azure-batch, azure-cosmos). In the case of batch, a similar set of modules is available in the main azure package (with some objects changed), but the functionality of azure-cosmos is separate. Their organizational model is not exactly straightforward, but using azure and azure-cosmos appears to function, while azure-batch and azure-cosmos conflict.

File: oliver/azbatch.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        stream = os.popen(
+            "az batch account keys list --name "
+            + batch_name
+            + " --resource-group "
+            + resource_group
+        )
+        key = stream.read()
+        key = json.loads(key)
+        self.key = key["primary"]
+        stream = os.popen(
```
Comment (claymcleod) [inline oliver/azbatch.py]: Is there an API we can call rather than shelling out here?

File: scripts/oliver [hunk 2]
```diff
@@ -28,6 +29,7 @@ SUBCOMMANDS = {
     "aws": aws,
     "azure": azure,
     "aggregate": aggregate,
+    "azbatch": azbatch,
```
Comment (claymcleod) [inline scripts/oliver]: I think this can now live under the Azure subcommand right?

File: oliver/azbatch.py (hunk 1 above)
Comment (claymcleod) [reply oliver/azbatch.py]: Okay, I’m good with whatever you decide.

File: scripts/oliver (hunk 2 above)
Comment (adthrasher) [reply scripts/oliver]: Yes, it should have been removed.

**Pull request closed without merging**